### PR TITLE
created pre-commit config file with linter hooks

### PR DESCRIPTION
**Pull request by rishavanand**

### CODE REVIEW

**Comment by rishavanand** [comment]
Lines hooks have been added but not yet run.

**Comment by lachmanfrantisek** [comment]
> Currently I'm working on passing the dockerfile lint. It requires the `Label (Name and Version)`.
> Is it acceptable?

I am not sure if want to change the building of the images.

@sakalosj does this change break anything? (What about skipping that for now and create a new issue for that?)

**Comment by rishavanand** [comment]
continuing on #14

**Comment by rishavanand** [comment]
@csomh Following from (#14) The pre commit config file has already been merged (#14) so I'll rebase this (#17) and add more hooks/linters. The first linter I'll add is `jinjalint` and the second one is the `dockerfile_lint`. Could you clarify [#issuecomment-593147198](#issuecomment-593147198) from above? Also do you have any more hooks in mind?

**Comment by sakalosj** [comment]
> > Currently I'm working on passing the dockerfile lint. It requires the `Label (Name and Version)`.
> > Is it acceptable?
> 
> I am not sure if we want to change the building of the images.
> 
> @sakalosj does this change break anything? (What about skipping that for now and create a new issue for that?)

no, it will not break anything

**Comment by csomh** [comment]
Hey @rishavanand :wave: 

What would be the additional benefit that `jinjalint` would bring and `prettier` doesn't have?

Let's not add `dockerfile_lint`: our projects don't have such complex Dockerfiles and they don't change that often, so it would be an overkill to check them on each commit.

**Comment by rishavanand** [comment]
Hi @csomh 

I'll give an example:
```
<h2>
    Hello from Rishav :)
    {% if false %}
</h2>
{% endif %}
```
Here the closing h2 tag will not be rendered since the condition is always false.  Such logical mistakes are covered by `jinjalint` but not by `prettier`.

**Comment by csomh** [comment]
Makes sense @rishavanand, let's have `jinjalint`! :man_dancing:

**Comment by rishavanand** [comment]
@csomh I've added the `jinjalint` hook. Please check.

@csomh @lachmanfrantisek Also let me know if I can run `pre-commit run --all-files` and fix the errors/warnings. It was discussed in #14 . We can create a new issue for this.

**Comment by lachmanfrantisek** [comment]
> @csomh @lachmanfrantisek Also let me know if I can run `pre-commit run --all-files` and fix the errors/warnings. It was discussed in #14 . We can create a new issue for this.

At least, use a specific commit for those changes. (I am ok with having those changes in this PR.)

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -38,4 +44,4 @@ repos:
     hooks:
       - id: check-rebase
         args:
-          - git://github.com/packit-service/dashboard.git
+          - git://github.com/packit-service/dashboard.git
```
Comment (csomh) [inline .pre-commit-config.yaml]: Let's make sure there is a newline at the end of the file.

**Comment by csomh** [comment]
@rishavanand this looks good!

Could you please squash (and maybe also rebase) your commits? The current change doesn't need 3 commits (it just adds a new hook).

File: .pre-commit-config.yaml (hunk 1 above)
Comment (rishavanand) [reply .pre-commit-config.yaml]: Done!

**Comment by rishavanand** [comment]
@csomh Squashed & rebased. I hope I did it as expected.

**Comment by rishavanand** [comment]
@lachmanfrantisek okay, I'll run `pre-commit` for all files and create a new commit here.

**Comment by jpopelka** [comment]
> Squashed & rebased. I hope I did it as expected.

It's been squashed, but the rebase is not OK.
Did you run `git rebase`? What does `git remote -v` show?

**Comment by rishavanand** [comment]
@jpopelka I was a bit confused. I did the rebase this time. I hope it is fine now. This is what `git remote -v` outputs on my system now:
```
origin	git@github.com:rishavanand/dashboard.git (fetch)
origin	git@github.com:rishavanand/dashboard.git (push)
upstream	git@github.com:packit-service/dashboard.git (fetch)
upstream	git@github.com:packit-service/dashboard.git (push)
```

**Comment by rishavanand** [comment]
@lachmanfrantisek I was running that `pre-commit` for all files and found that prettier does not work properly for jinja templates. For example:

This is what prettier generates:
```
{% for build in builds_list %}
<tr role="row">
  <td role="cell" data-label="Project">{{ build["project"] }}</td>
</tr>
{% endfor %}
```

This is what it should look like as per jinja indentation:
```
{% for build in builds_list %}
  <tr role="row">
    <td role="cell" data-label="Project">{{ build["project"] }}</td>
  </tr>
{% endfor %}
```

This is making the `jinjalint` hook throw an indentation error. What do you suggest?

**Comment by csomh** [comment]
@rishavanand is there an option for jinjalint to disable indentation-checks?

**Comment by rishavanand** [comment]
@csomh found an argument `--parse-only` which skips the indentation-check 😄 It's wasn't mentioned in their readme though 😞

**Comment by lachmanfrantisek** [comment]
> @lachmanfrantisek I was running that `pre-commit` for all files and found that prettier does not work properly for jinja templates. For example:

Another option is to ignore prettier for the HTML files and use only the jinjalint for that.

**Comment by csomh** [comment]
> > @lachmanfrantisek I was running that `pre-commit` for all files and found that prettier does not work properly for jinja templates. For example:
> 
> Another option is to ignore prettier for the HTML files and use only jinjalint for that.

I'm very much against this: jinjalint is a prototype project, not updated since 2018, plus it's only a linter (forcing us to do tedious manual work) instead of being a formatter (doing work instead of us).

**Comment by rishavanand** [comment]
@csomh Rightly said `jinjalint` is only a linter. It only checks for errors and does not correct them. So we can use it for finding logical errors and skip for indentation. (more on #26 )

**Comment by lachmanfrantisek** [comment]
> Another option is to ignore prettier for the HTML files and use only jinjalint for that.

OK, I thought it's also the reformatter... Then it does not make sense.

Let's continue with the discussion in the other PR to not need to have two similar ones in parallel..;)

**Pull request merged**